pp.c

```diff
@@ -3522,9 +3522,11 @@ PP(pp_index)
     if (PL_op->op_private & OPpTRUEBOOL) {
         PUSHs( ((retval != -1) ^ cBOOL(PL_op->op_private & OPpINDEX_BOOLNEG))
                     ? &PL_sv_yes : &PL_sv_no);
-        if (PL_op->op_private & OPpTARGET_MY)
+        if (PL_op->op_private & OPpTARGET_MY) {
             /* $lex = (index() == -1) */
             sv_setsv(TARG, TOPs);
+            SETTARG;
```

Comment:
Isn't the `SETs(TARG)` part of `SETTARG` redundant when we've just done `sv_setsv(TARG, TOPs)`?
I.e. couldn't this just be `SvSETMAGIC(TARG)`?

Reply:
Or just call `sv_setsv_mg(TARG, TOPs)`?

Reply:
That would break the fix for returning the lexical, see #17739 

The code could be clearer though, to avoid pushing the wrong SV, I'll re-work this.

Reply:
@ilmari, @leont, do you consider this resolved now?